File: capy/src/e2e-cfun/java/dev/capylang/test/OverloadResolutionTest.java
package dev.capylang.test;

import org.junit.jupiter.api.Test;

import static org.assertj.core.api.Assertions.assertThat;

class OverloadResolutionTest {
    @Test
    void shouldPreferExactListOverloadOverWidenableOne() {
        assertThat(OverloadResolution.chooseInts()).isEqualTo(103);
        assertThat(OverloadResolution.chooseLongs()).isEqualTo(203);
    }
}

File: capy/src/e2e-cfun/js/dev/capylang/test/OverloadResolution.test.js
const test = require('node:test');
const { assert, generatedModule } = require('./_support');

test('OverloadResolution', () => {
    const overload = generatedModule('dev/capylang/test/OverloadResolution.js');
    assert.equal(overload.chooseInts(), 103);
    assert.equal(overload.chooseLongs(), 203);
});
